sonar-python-plugin/src/main/java/org/sonar/plugins/python/indexer/SonarLintPythonIndexer.java

```diff
@@ -73,6 +79,14 @@ public void buildOnce(SensorContext context) {
     globalSymbolsStep.execute(files, context);
   }
 
+  // SonarLintCache has to be set lazily because SonarLintPythonIndex is injected in the PythonSensor
+  @Override
+  public void setSonarLintCache(@Nullable SonarLintCache sonarLintCache) {
+    if (sonarLintCache != null) {
```

Comment:
In practice, `sonarLintCache` will never be null here, right?
It's only because we defined the method at `PythonIndexer` level and pass whatever is provided to the sensor constructor that we don't make a distinction?
Should we have a comment explaining this is defensive? Or is there a way we can safely avoid having the defensive code altogether?

Reply:
A comment has been added that explains that this check is defensive and should never fail in practice.

As an improvement, changes could be made in a future PR by for example directly injecting SonarLintCache into SonarLintPythonIndexer instead of PythonSensor.